packages/functional-tests/pages/products/index.ts

```diff
@@ -70,8 +82,37 @@ export class SubscribePage extends BaseLayout {
     return msg;
   }
 
-  discountAppliedSucess() {
-    return this.page.innerText('.coupon-header');
+  async discountAppliedSuccess() {
+    const discount = this.page.locator('.coupon-header:has-text("Discount Reward Applied")');
+    await discount.waitFor();
+    return discount.isVisible();
+  }
+
+  async oneTimeDiscountSuccess() {
+    const discount = this.page.locator
+      ('[data-testid="coupon-success"]:has-text("Your plan will automatically renew at the list price.")');
+    await discount.waitFor();
+    return discount.isVisible();
+  }
+
+  async discountListPrice() {
+    const listPrice = this.page.locator('.plan-details-total:has-text("List Price")');
+    return listPrice.isVisible();
```

Comment:
Should this have a `listPrice.waitFor()` above this?

Reply:
adding waitfor() for the discountListPrice() and discountLineItem() were timing out because it kept on waiting for the elements while asserting for a negative scenario for test "remove a coupon and verify".